release/air_tests/air_benchmarks/workloads/torch_benchmark.py

```diff
@@ -0,0 +1,298 @@
+        f"Avg loss: {test_loss:>8f} \n"
+    )
+    return test_loss
+
+
+def train_func(use_ray: bool, config: Dict):
+    if use_ray:
+        from ray.air import session
+        import ray.train as train
+
+    batch_size = config["batch_size"]
+    lr = config["lr"]
+    epochs = config["epochs"]
+
+    if use_ray:
+        world_size = session.get_world_size()
+    else:
+        world_size = distributed.get_world_size()
+
+    worker_batch_size = batch_size // world_size
+
+    # Load datasets. Use download=False to catch errors in preparation, as the
+    # data should have already been downloaded.
+    training_data = datasets.FashionMNIST(
+        root="/tmp/data_fashion_mnist",
+        train=True,
+        download=False,
+        transform=ToTensor(),
+    )
+
+    test_data = datasets.FashionMNIST(
+        root="/tmp/data_fashion_mnist",
+        train=False,
+        download=False,
+        transform=ToTensor(),
+    )
+
+    # Create data loaders.
+    train_dataloader = DataLoader(training_data, batch_size=worker_batch_size)
+    test_dataloader = DataLoader(test_data, batch_size=worker_batch_size)
+
+    # Create model.
+    model = NeuralNetwork()
+
+    if use_ray:
+        model = train.torch.prepare_model(model)
+    else:
+        model = nn.parallel.DistributedDataParallel(model)
+
+    loss_fn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=lr)
+
+    for _ in range(epochs):
+        train_epoch(train_dataloader, model, loss_fn, optimizer, world_size=world_size)
+        loss = validate_epoch(test_dataloader, model, loss_fn, world_size=world_size)
+        if use_ray:
+            session.report(dict(loss=loss))
+        else:
+            print(f"Reporting loss: {loss:.4f}")
+
+
+def train_torch_ray_air(*, config: dict, num_workers: int = 4, use_gpu: bool = False):
+    # This function is kicked off by the main() function and runs a full training
+    # run using Ray AIR.
+    from ray.train.torch import TorchTrainer
+
+    def train_loop(config):
+        train_func(use_ray=True, config=config)
+
+    trainer = TorchTrainer(
+        train_loop_per_worker=train_loop,
+        train_loop_config=config,
+        scaling_config={"num_workers": num_workers, "use_gpu": use_gpu},
+    )
+    result = trainer.fit()
+    print(f"Last result: {result.metrics}")
+
+
+def train_torch_vanilla_worker(
+    *, config: dict, rank: int, world_size: int, master_addr: str, master_port: int
+):
+    # This function is kicked off by the main() function and runs the vanilla
+    # training script on a single worker.
+    os.environ["MASTER_ADDR"] = master_addr
+    os.environ["MASTER_PORT"] = str(master_port)
+    distributed.init_process_group("gloo", rank=rank, world_size=world_size)
+
+    train_func(use_ray=False, config=config)
+
+    distributed.destroy_process_group()
+
+
+def train_torch_vanilla(*, config: dict, num_workers: int = 4, use_gpu: bool = False):
```

Comment:
Seems like `use_gpu` is being ignored for the vanilla flow - should we remove it for now?

Reply:
Let's leave it since GPU is an exit criteria for this benchmark (to be done immediately); we can either merge it in as-is or wait until the GPU benchmark is added too.